autorag/nodes/queryexpansion/query_decompose.py

```diff
@@ -0,0 +1,93 @@
+import asyncio
+from typing import List
+
+from llama_index.llms.llm import BaseLLM
+
+decompose_prompt = """Decompose a question in self-contained sub-questions. Use \"The question needs no decomposition\" when no decomposition is needed.
+
+    Example 1:
+
+    Question: Is Hamlet more common on IMDB than Comedy of Errors?
+    Decompositions: 
+    1: How many listings of Hamlet are there on IMDB?
+    2: How many listing of Comedy of Errors is there on IMDB?
+
+    Example 2:
+
+    Question: Are birds important to badminton?
+
+    Decompositions:
+    The question needs no decomposition
+
+    Example 3:
+
+    Question: Is it legal for a licensed child driving Mercedes-Benz to be employed in US?
+
+    Decompositions:
+    1: What is the minimum driving age in the US?
+    2: What is the minimum age for someone to be employed in the US?
+
+    Example 4:
+
+    Question: Are all cucumbers the same texture?
+
+    Decompositions:
+    The question needs no decomposition
+
+    Example 5:
+
+    Question: Hydrogen's atomic number squared exceeds number of Spice Girls?
+
+    Decompositions:
+    1: What is the atomic number of hydrogen?
+    2: How many Spice Girls are there?
+
+    Example 6:
+
+    Question: {question}
+
+    Decompositions:"
+    """
+
+
+def query_decompose(queries: List[str], llm: BaseLLM,
+                    prompt: str = decompose_prompt) -> List[List[str]]:
+    """
+    decompose query to little piece of questions.
+    :param queries: List[str], queries to decompose.
+    :param llm: BaseLLM, language model to use.
+    :param prompt: str, prompt to use for query decomposition.
+        default prompt is guidelines into simpler sub-questions or stating no decomposition is needed, illustrated with examples.
+    :return: List[List[str]], list of decomposed query. Return input query if query is not decomposable.
+    """
+    # Run async query_decompose_pure function
+    tasks = [query_decompose_pure(query, llm, prompt) for query in queries]
+    loop = asyncio.get_event_loop()
+    results = loop.run_until_complete(asyncio.gather(*tasks))
+    return results
+
+
+async def query_decompose_pure(query: str, llm: BaseLLM,
+                               prompt: str = decompose_prompt) -> List[str]:
+    """
+    decompose query to little piece of questions.
+    :param query: str, query to decompose.
+    :param llm: BaseLLM, language model to use.
+    :param prompt: str, prompt to use for query decomposition.
+        default prompt is guidelines into simpler sub-questions or stating no decomposition is needed, illustrated with examples.
```

Comment:
Its prompt comes from Visconde's StrategyQA few-shot prompt.
Please mention it properly.